Source/Core/VideoCommon/TextureConversionShader.cpp

```diff
@@ -126,9 +140,20 @@ static void WriteSampleColor(char*& p, const char* colorComp, const char* dest,
 	}
 	else
 	{
-		WRITE(p, "  %s = Tex0.Sample(samp0, uv0 + float2(%d, 0) * sample_offset).%s;\n",
-			dest, xoffset, colorComp
-			);
+		if (numInputSamples == 1)
+		{
+			WRITE(p, "  %s = Tex0.Sample(samp0, uv0 + float2(%d, 0) * sample_offset).%s;\n",
+				dest, xoffset, colorComp
+				);
+		}
+		else
+		{
+			WRITE(p, "	texcol = 0;\n");
+			WRITE(p, "  for(int i = 0; i < %d; ++i)\n", numInputSamples);
+			WRITE(p, "		texcol += Tex0.Load(int3((uv0.x+%d*sample_offset)*(width), uv0.y*(height), 0), i);\n", xoffset);
+			WRITE(p, "	texcol /= %d;\n", numInputSamples);
+			WRITE(p, "	%s = texcol.%s;\n", dest, colorComp);
```

Comment:
This isn't going to do the right thing in scaleByHalf mode, which requires linear sampling.  Granted, it might not matter; we don't really do downscaling correctly anyway outside of 1x resolution.